typescript/hardhat/src/TestAbacusDeploy.ts

```diff
@@ -143,62 +143,71 @@ export class TestAbacusDeploy extends TestDeploy<
     return this.instances[domain].validatorManager;
   }
 
-  async processMessages() {
-    await Promise.all(
-      this.domains.map((d) => this.processMessagesFromDomain(d))
-    );
+  async processMessages(): Promise<
```

Comment:
There's some interesting, possibly confusing behavior here, in that the domain order matters.

For example, imagine I have an unprocessed message 1 with destination A, and when it is processed, it dispatches message 2 to B.

If the domain order is [A, B], then `processMessages` will wind up processing 1 and 2, but if the domain order is [B, A] it will only process 1.

I wonder if `processMessages` should first checkpoint on every domain, and *then* process all unprocessed messages with index < checkpointed index.

Does this make sense? If so, can you follow a "bug" ticket?

Reply:
https://github.com/abacus-network/abacus-monorepo/issues/337